src/Univapay/Resources/ThreeDSMPI.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Univapay\Resources;
+
+use Univapay\Enums\Reason;
+use Univapay\Utility\FunctionalUtils;
+use Univapay\Errors\UnivapayLogicError;
+
+class ThreeDSMPI
+{
+    use Jsonable;
+
+    public $authenticationValue;
+    public $eci;
+    public $dsTransactionId;
+    public $serverTransactionId;
+    public $messageVersion;
+    public $transactionStatus;
+
+    public function __construct(
+        $authenticationValue,
+        $eci,
+        $dsTransactionId,
+        $serverTransactionId,
+        $messageVersion,
+        $transactionStatus
+    ) {
+        $this->authenticationValue = $authenticationValue;
+        $this->eci = $eci;
+        $this->dsTransactionId = $dsTransactionId;
+        $this->serverTransactionId = $serverTransactionId;
+        $this->messageVersion = $messageVersion;
+        $this->transactionStatus = $transactionStatus;
+
+        $this->validate();
+    }
+
+    private function validate()
+    {
+        $allNullOrEmpty = $this->isAllNullOrEmpty();
+
+        // not using 3DS MPI
+        if ($allNullOrEmpty) {
+            return;
+        }
```

Comment:
This is fine, but I wonder what is the point of creating an empty object... We could enforce that the user just set null instead for an empty object and then we can eliminate this extra check here